crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -6,41 +8,88 @@ use super::{
     text_index::{TextIndex, TextRange},
 };
 
+/// A [`NodePtr`] that points to a [`RuleNode`].
 #[derive(Clone, Debug, PartialEq, Eq)]
-struct CursorPathElement {
+struct RuleNodePtr {
```

Comment:
I prefer the original name because it documents the purpose of the struct, rather than merely re-describing what it contains.

Reply:
In general I agree that naming should help document the purpose, but in this case I believe that separating where it's used versus how it's used makes the code below slightly less obvious.

Due to the way that the cursor is implemented, the Cursor path element has to contain what the pointed to node has (child number and text offset) and the only way these path elements are actually used in the code is to convert between the `NodePtr` <-> this struct.

Because of this, I think it's better to emphasise that that the conversion is almost 1-1, with the rule node being more specific type. Otherwise reading `NodePtr::to_path_element` or `PathElement::into_node_ptr` might imply a less direct conversion.

All in all, this simplifies the model to "Cursor iteration always needs (node, child_id, offset)". The fact that the path can only contain rule nodes is external to this module and also used in the public  `fn path_rule_nodes`.

Maybe I can change it to `PathRuleNode{Ptr,Pointer}`?

Reply:
happy with `PathRuleNode`

Reply:
Done in 03d055a13bd4ce7e4127442a290a766538597b2c